notebook/services/api/handlers.py

```diff
@@ -18,6 +18,13 @@ def get(self):
         self.set_header('Content-Type', 'text/x-yaml')
         return web.StaticFileHandler.get(self, 'api.yaml')
 
+    @web.authenticated
+    def options(self, *args, **kwargs):
```

Comment:
ah, sorry for the confusion. This is just a super-bad choice of names. This isn't be base APIHandler class, this is a weird, only partially implemented handler for providing the REST API spec. The base class is in base/handlers.py